File: clients/rust/model/src/game.rs
use crate::*;
#[derive(Clone, Debug, trans::Trans)]
pub struct Game {
    pub current_tick: i32,
    pub properties: Properties,
    pub level: Level,
    pub players: Vec<Player>,
    pub units: Vec<Unit>,
    pub bullets: Vec<Bullet>,
    pub mines: Vec<Mine>,
    pub loot_boxes: Vec<LootBox>,
}

File: clients/haskell/src/RAIC/Model/Game.hs
module RAIC.Model.Game where

import           GHC.Generics          (Generic)
import           RAIC.Model.Bullet     (Bullet)
import           RAIC.Model.Level      (Level)
import           RAIC.Model.LootBox    (LootBox)
import           RAIC.Model.Mine       (Mine)
import           RAIC.Model.Player     (Player)
import           RAIC.Model.Properties (Properties)
import           RAIC.Model.Unit       (Unit)
import           RAIC.StreamWrapper    (Trans)

data Game = Game {
  current_tick :: Int,
  properties   :: Properties,
  level        :: Level,
  players      :: [Player],
  units        :: [Unit],
  bullets      :: [Bullet],
  mines        :: [Mine],
  loot_boxes   :: [LootBox]
} deriving (Generic, Show)

instance Trans Game
